contracts/voting-verifier/src/contract/execute.rs

```diff
@@ -130,12 +139,15 @@ pub fn verify_messages(
                 &message.hash(),
                 &state::PollContent::<Message>::new(message.clone(), id, idx),
             )
-            .map_err(ContractError::from)?;
+            .change_context(ContractError::StorageError)?;
     }
 
     let messages = msgs_to_verify
         .into_iter()
-        .map(|msg| (msg, &config.msg_id_format).try_into())
+        .map(|msg| {
+            TxEventConfirmation::try_from((msg.clone(), &config.msg_id_format))
+                .map_err(|err| report!(err))
```

Comment:
just curious, have you ever tried if `map_err(report!)` works?

Reply:
Doesn't seem to:
```
150 |                 .map_err(report!)
    |                                 ^ expected one of `(`, `[`, or `{`
```